source/common/router/config_impl.h

```diff
@@ -54,6 +54,8 @@ class SslRedirector : public DirectResponseEntry {
   // Router::DirectResponseEntry
   std::string newPath(const Http::HeaderMap& headers) const override;
   Http::Code responseCode() const override { return Http::Code::MovedPermanently; }
+  Optional<std::string> responseBody() const override { return Optional<std::string>(); }
```

Comment:
Any reason to not kill off the optional and insead return EMPTY_STRING by default?  Especially as on config parsing we return an empty optional if the body is empty in any case.